src/pages/ForgotPassword/ForgotPassword.tsx

```diff
@@ -1,9 +1,195 @@
-import { Box, Grid, Typography, colors } from '@mui/material';
+import { useEffect, useState } from 'react';
+import {
+  Box,
+  Grid,
+  Typography,
+  TextField,
+  Modal,
+  colors,
+  Divider
+} from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+import OtpInput from 'react-otp-input';
+
 import loginImg from '../../images/login.png';
+import forgotPasswordImg from '../../images/forgot-password.png';
+import { CustomButton } from '../../components/CustomButton';
+import { useClock } from '../../hooks/useClock';
 
 const ForgotPassword = () => {
+  const [otp, setOtp] = useState<string>('');
+  const [isShowResendOtp, setIsShowResendOtp] = useState<boolean>(false);
+  const { time, setTime } = useClock({ hours: 0, minutes: 0, seconds: 0 });
+  const [open, setOpen] = useState<boolean>(false);
+  const [disabled, setDisabled] = useState<boolean>(true);
+
+  const handleOpenModal = () => {
+    setTime({ hours: 0, minutes: 2, seconds: 0 });
+    setOpen(true);
+  };
+
+  const handleCloseModal = () => setOpen(false);
+
+  const handleChangeOtp = (otp: string) => {
+    setOtp(otp);
+  };
+
+  const resendOtp = () => {
+    setTime({ hours: 0, minutes: 2, seconds: 0 });
+    setIsShowResendOtp(false);
+    setOtp('');
+  };
+
+  useEffect(() => {
+    if (otp.length === 6) {
+      setDisabled(false);
+    } else {
+      setDisabled(true);
+    }
+  }, [otp]);
+
+  useEffect(() => {
+    if (time === '00:00:00' && open === true) {
+      setIsShowResendOtp(true);
+    }
+  }, [time, open]);
+
   return (
     <Grid container>
+      <Modal open={open}>
```

Comment:
Dùng component Dialog. Tách toàn bộ Dialog OTP này ra 1 component OTPInput để tái sử dụng ở nhiều màn hình khác